pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,169 @@
 package adm
 
 import (
-	"context"
-	"fmt"
+	"bytes"
+	"io"
+	"net/http"
 	"testing"
 
 	"github.com/codeready-toolchain/toolchain-common/pkg/test"
-	"github.com/kubesaw/ksctl/pkg/configuration"
 	clicontext "github.com/kubesaw/ksctl/pkg/context"
 	. "github.com/kubesaw/ksctl/pkg/test"
 
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
-	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest/fake"
+	cgtesting "k8s.io/client-go/testing"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	"k8s.io/kubectl/pkg/scheme"
 )
 
 func TestRestartDeployment(t *testing.T) {
 	// given
```

Comment:
Correct me if I'm wrong, but I haven't found anywhere in the tests that the pods were really deleted and the deployment was modified. Could you add some interceptor for this?

Reply:
added  for deleted pods, 
as for deployment changes, i couldn't get a fakeclient out of fake restclient created where we can check the deployments. since the clients are different , i am not able to check the changes in deployments